controller/identity/validator.go

```diff
@@ -0,0 +1,98 @@
+package identity
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/linkerd/linkerd2/pkg/identity"
+	kauthnApi "k8s.io/api/authentication/v1"
+	kauthzApi "k8s.io/api/authorization/v1"
+	"k8s.io/apimachinery/pkg/util/validation"
+	k8s "k8s.io/client-go/kubernetes"
+	kauthn "k8s.io/client-go/kubernetes/typed/authentication/v1"
+	kauthz "k8s.io/client-go/kubernetes/typed/authorization/v1"
+)
+
+// K8sTokenValidator implements Validator for Kubernetes bearer tokens.
+type K8sTokenValidator struct {
+	authn  kauthn.AuthenticationV1Interface
+	domain *TrustDomain
+}
+
+// NewK8sTokenValidator takes a kubernetes client and trust domain to create a
+// K8sTokenValidator.
+//
+// The kubernetes client is used immediately to validate that the client has
+// sufficient privileges to perform token reviews. An error is returned if this
+// access check fails.
+func NewK8sTokenValidator(
+	k8s k8s.Interface,
+	domain *TrustDomain,
+) (*K8sTokenValidator, error) {
+	if err := checkAccess(k8s.AuthorizationV1()); err != nil {
+		return nil, err
+	}
+
+	authn := k8s.AuthenticationV1()
```

Comment:
We could have this guy be declared at the beginning of this function to avoid calling `AuthenticationV1()` twice.

Reply:
One is `AuthorizationV1` and the other is `AuthenticationV1`